apps/web/src/components/templates/ExecutionDetailsModalWrapper.tsx

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from 'react-query';
+import { getActivityList } from '../../api/activity';
+import { ExecutionDetailsModal } from '../activity/ExecutionDetailsModal';
+import { LoadingOverlay, useMantineTheme } from '@mantine/core';
+import { colors } from '../../design-system';
+
+interface Props {
+  transactionId: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export const ExecutionDetailsModalWrapper = ({ transactionId, isOpen, onClose }: Props) => {
+  const theme = useMantineTheme();
+  const {
+    data: notification,
+    isLoading,
+    isFetching,
+  } = useQuery<{ data: any[] }>(['activitiesList', transactionId], () => getActivityList(0, { transactionId }), {
+    enabled: transactionId.length > 0,
+  });
+
+  return (
+    <>
+      <LoadingOverlay
+        visible={isLoading || isFetching}
+        overlayColor={theme.colorScheme === 'dark' ? colors.B30 : colors.B98}
+        loaderProps={{
+          color: colors.error,
+        }}
+      />
+      {notification?.data?.length && (
+        <ExecutionDetailsModal
+          notificationId={notification?.data[0]._id}
+          modalVisibility={isOpen}
+          onClose={onClose}
+          origin={location.pathname}
```

Comment:
Need to add a link text in the original component, and not only the location path. Did not want to create conflicts for now

Reply:
As you did a wrapper I do not think you would create conflicts with my changes on the design implementation. I think you should be good.